lib/spack/spack/spec.py

```diff
@@ -3384,7 +3384,8 @@ def __init__(self, spec, hash):
 class NoSuchHashError(SpecError):
     def __init__(self, hash):
         super(NoSuchHashError, self).__init__(
-            "No installed spec matches the hash: '%s'")
+            "No installed spec matches the hash: '%s'"
+        % hash)
```

Comment:
Flake8 doesn't like your indentation:
```
lib/spack/spack/spec.py:3388: [E122] continuation line missing indentation or outdented
```
Might need to use parentheses? I've always just used `"{0}".format(foo)` instead of `"%s" % foo`.